cranelift/codegen/src/ir/pcc.rs

```diff
@@ -806,6 +1186,32 @@ impl<'a> FactContext<'a> {
                 };
                 Ok(specific_ty_and_offset)
             }
+            Fact::DynamicMem {
+                ty,
+                min: _,
+                max:
+                    Expr {
+                        base: BaseExpr::GlobalValue(max_gv),
+                        offset: max_offset,
+                    },
+                nullable: _,
+            } => match &self.function.memory_types[*ty] {
+                ir::MemoryTypeData::DynamicMemory {
+                    gv,
+                    size: mem_static_size,
+                } if gv == max_gv => {
+                    let end_offset = max_offset
+                        .checked_add(i64::from(size))
+                        .ok_or(PccError::Overflow)?;
+                    ensure!(
+                        end_offset
+                            <= i64::try_from(*mem_static_size).map_err(|_| PccError::Overflow)?,
```

Comment:
Style nitpick: might be easier to read if the `try_from` expression was pulled out to its own variable, similar to how the `end_offset` is.

Reply:
Done!